src/assets/scripts/client/aircraft/FlightManagementSystem/LegModel.js

```diff
@@ -1,538 +1,780 @@
-     * the output will not be valid for anything that works with a routeString
+     * @for RouteModel
+     * @property isStarLeg
+     * @type {boolean}
+     */
+    get isStarLeg() {
+        return this.isProcedureLeg && this._procedureModel.procedureType === PROCEDURE_TYPE.STAR;
+    }
+
+    /**
+     * Returns the type of this leg
      *
-     * @property procedureAndExitName
+     * @for LegModel
+     * @property legType
      * @type {string}
-     * @return {string}
      */
-    get procedureAndExitName() {
-        if (!this.isProcedure) {
-            return null;
-        }
-
-        return `${this._procedureRouteModel.procedure}.${this._procedureRouteModel.exit}`;
+    get legType() {
+        return this._legType;
     }
 
     /**
-     * The active `WaypointModel`.
+     * Returns the `WaypointModel` immediately following the `#currentWaypoint`
      *
-     * Assumed to always be the first item in
-     * the `#waypointCollection`
-     *
-     * @property currentWaypoint
-     * @return {WaypointModel}
+     * @for LegModel
+     * @property nextWaypoint
+     * @type {WaypointModel}
      */
-    get currentWaypoint() {
-        return this.waypointCollection[0];
+    get nextWaypoint() {
+        return this._waypointCollection[1];
     }
 
     /**
-     * The `WaypointModel` immediately following the `#currentWaypoint`
-     * in the flightPlan
+     * Returns the route string for this leg
      *
-     * Used when calculating headings to the next waypoint.
+     * @for LegModel
+     * @property routeString
+     * @type {string}
+     */
+    get routeString() {
+        return this._routeString;
+    }
+
+    /**
+     * Return the `#_waypointCollection`
      *
-     * @property nextWaypoint
-     * @return {WaypointModel}
+     * @for LegModel
+     * @property waypoints
+     * @type {array<WaypointModel>}
      */
-    get nextWaypoint() {
-        return this.waypointCollection[1];
+    get waypoints() {
+        return this._waypointCollection;
     }
 
+    // ------------------------------ LIFECYCLE ------------------------------
+
     /**
-     * Instantiate the class properties
+     * Initialize class properties
      *
      * @for LegModel
      * @method init
-     * @param routeSegment {string}
-     * @param runway {string}
-     * @param holdWaypointProps {object}
+     * @param navigationLibrary {NavigationLibrary}
+     * @param routeString {string}
+     * @chainable
      */
-    init(routeSegment, runway, flightPhase, holdWaypointProps) {
-        this.isProcedure = RouteModel.isProcedureRouteString(routeSegment);
-        // TODO: replace with constant
-        this._isHold = RouteModel.isHoldRouteString(routeSegment) || routeSegment === 'GPS';
+    init(navigationLibrary, routeString) {
+        this._routeString = routeString;
+
+        const [entryOrFixName, airwayOrProcedureName, exit] = routeString.split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
```

Comment:
don't we have a helper function/method for this somewhere?  If not, this logic _should_ be abstracted so it can be tested.  This logic should not be tied to a particular method like this, even though it is pretty simple, it's also very important and could potentially be used elsewhere.